src/npcmove.cpp

```diff
@@ -1669,12 +1662,12 @@ void npc::alt_attack(int target)
         std::vector<point> trajectory;
         int linet, light = g->light_level();
 
-        if (dist <= confident_range(weapon_index) && wont_hit_friend(tarx, tary, weapon_index)) {
+        if (dist <= confident_range(weapon_index) && wont_hit_friend( tar, weapon_index )) {
 
-            if (g->m.sees(posx(), posy(), tarx, tary, light, linet)) {
-                trajectory = line_to(posx(), posy(), tarx, tary, linet);
+            if (g->m.sees(posx(), posy(), tar.x, tar.y, light, linet)) {
+                trajectory = line_to(posx(), posy(), tar.x, tar.y, linet);
             } else {
-                trajectory = line_to(posx(), posy(), tarx, tary, 0);
+                trajectory = line_to(posx(), posy(), tar.x, tar.y, 0);
```

Comment:
Added the new behavior later when I figured out that it's needed. Forgot to update this case (and possibly some other).
`map::sees` for tripoints currently only checks 2D `sees` and sets the second bresenham to 0.